File: Class/1.rb
class Cal
  # 생성자 initialize
  def initialize(v1,v2)
    # @v1 = 인스턴스 변수
    # v1 = 로컬변수
    @v1 = v1
    @v2 = v2
  end
  def add()
    return @v1 + @v2
  end
  def subtract()
    return @v1 - @v2
  end
end

# 인스턴스를 생성
c1 = Cal.new(10,10)
# 클래스 안의 메소드를 사용하기
p c1.add()
p c1.subtract()
p Cal.new(10000, 1).add()

c2 = Cal.new(30,20)
p c2.add()
p c2.subtract()

File: Class/1.py
# -*- coding: utf-8 -*-
# 칼이 대문자 아니어도 됨
class Cal(object):
    # 메소드를 호출할때 메소드의
    # 첫번째의 매개변수가
    # 인스턴스 변수가 된다
    # 그걸 사용함으로써 클래스 전체에서
    # 인스턴스 변수를 사용할수 있다.
    def __init__(self, v1, v2):
        self.v1 = v1
        self.v2 = v2
    def add(self):
        return self.v1 + self.v2
    def subtract(self):
        return self.v1 - self.v2

# 앞변수가 v2 뒷변수가 v1의 매개변수값이 된다.
c1 = Cal(10,10)
print (c1.add())
print (c1.subtract())

c2 = Cal(30,20)
print (c2.add())
print (c2.subtract())
